src/sources/socket/mod.rs

```diff
@@ -1083,6 +1134,42 @@ mod test {
         .await;
     }
 
+    // TODO: maybe this should be in an integration test, such as `src/sources/redis/mod.rs` and `scripts/integration/redis`
+    // and so? There currently are no integration tests for the socket source
+    #[tokio::test]
+    async fn udp_decodes_chunked_gelf_messages() {
```

Comment:
I don't know if this kind of test is okay to be done here. It is more like an integration test as the ones we have in `scripts/integration` and `tests/integration`. The problem is, there is not any sockets integration test right now and didn't know the best way to do this.

It is kind of weird to test the chunked gelf deserializing in this module... What do you think about this?
If its not such a big deal, maybe we can leave it here, I don't now

Reply:
Yeah, I see what you are getting at. It would be nice to have integration tests, but you are correct that we don't have any for `socket` yet. Another place we could put them is in `tests/integration`. Those are Rust-based "integration tests" that are more black box.

Reply:
It seems that there are no sink/source related tests in that directory. I think it feels better as of right now to leave these tests with the other socket tests because they are kind of related...

But I agree with you that these kind of tests are better placed in `tests/integration`, along with a few others in this file that are not from this PR. I think this change would be better placed in a refactor/tech debt PR